src/foundry/common/data/data.mjs/actorData.d.ts

```diff
@@ -0,0 +1,130 @@
+    required: true;
+    default: (data: unknown) => { name: string; img: string };
+  };
+  items: fields.EmbeddedCollectionField<typeof documents.BaseItem>;
+  effects: fields.EmbeddedCollectionField<typeof documents.BaseActiveEffect>;
+  folder: fields.ForeignDocumentField<{ type: typeof documents.BaseFolder }>;
+  sort: typeof fields.INTEGER_SORT_FIELD;
+  permission: typeof fields.DOCUMENT_PERMISSIONS;
+  flags: typeof fields.OBJECT_FIELD;
+}
+
+interface ActorDataBaseProperties {
+  /**
+   * The _id which uniquely identifies this Actor document
+   */
+  _id: string | null;
+
+  /**
+   * The name of this Actor
+   */
+  name: string;
+
+  /**
+   * An Actor subtype which configures the system data model applied
+   */
+  type: string;
+
+  /**
+   * An image file path which provides the artwork for this Actor
+   */
+  img?: string | null;
+
+  /**
+   * The system data object which is defined by the system template.json model
+   */
+  data: object;
+
+  /**
+   * Default Token settings which are used for Tokens created from this Actor
+   */
+  token: PrototypeTokenData;
+
+  /**
+   * A Collection of Item embedded Documents
+   */
+  items: EmbeddedCollection<ConfiguredDocumentClass<typeof documents.BaseItem>, ActorData>;
+
+  /**
+   * A collection of ActiveEffect embedded Documents
+   */
+  effects: EmbeddedCollection<ConfiguredDocumentClass<typeof documents.BaseActiveEffect>, ActorData>;
+
+  /**
+   * The _id of a Folder which contains this Actor
+   * @defaultValue `null`
+   */
+  folder: string | null;
+
+  /**
+   * The numeric sort value which orders this Actor relative to its siblings
+   * @defaultValue `0`
+   */
+  sort: number;
+
+  /**
+   * An object which configures user permissions to this Actor
+   * @defaultValue `{ default: CONST.ENTITY_PERMISSIONS.NONE }`
+   */
+  permission: Partial<Record<string, ValueOf<typeof CONST.ENTITY_PERMISSIONS>>>;
+
+  /**
+   * An object of optional key/value flags
+   * @defaultValue `{}`
+   */
+  flags: Record<string, unknown>;
+}
+
+type ActorDataBaseSource = PropertiesToSource<ActorDataBaseProperties>;
+type ActorDataProperties = ActorDataBaseProperties & ConfiguredData<'Actor'>;
+type ActorDataSource = ActorDataBaseSource & ConfiguredSource<'Actor'>;
+
+type DocumentDataConstructor = typeof DocumentData;
+
+interface ActorDataConstructor extends DocumentDataConstructor {
+  new (data?: DeepPartial<ActorDataSource>, document?: documents.BaseActor | null): ActorData;
+
+  defineSchema(): ActorDataSchema;
+
+  /**
+   * The default icon used for newly created Actor documents
+   * @defaultValue `"icons/svg/mystery-man.svg"`
+   */
+  DEFAULT_ICON: string;
+}
+
+/**
+ * The data schema for a Actor document.
+ * @see BaseActor
+ */
+export type ActorData = DocumentData<ActorDataSchema, ActorDataProperties, ActorDataSource, documents.BaseActor> &
+  ActorDataProperties & {
+    _initializeSource(data: DeepPartial<ActorDataSource>): ActorDataSource;
+  };
```

Comment:
`_initialize` is missing

Reply:
`_initialize` is inherited from `DocumentData`. In foundry it just inherits the doc string

Reply:
But it’s still overridden in ActorData, right? So we should add it here (for consistency)

Reply:
It is overriden, but I figured I don't really need to include it since there's no changes to the docstring. so for the purpose of typing it makes no real difference. I can add them though, no problem